src/service_daemon.rs

```diff
@@ -233,6 +235,26 @@ impl ServiceDaemon {
         self.send_cmd(Command::StopBrowse(ty_domain.to_string()))
     }
 
+    /// Starts querying for the ip addresses of a hostname.
+    ///
+    /// Returns a channel `Receiver` to receive events about the hostname.
+    /// The caller can call `.recv_async().await` on this receiver to handle events in an
+    /// async environment or call `.recv()` in a sync environment.
+    pub fn resolve(&self, hostname: &str) -> Result<Receiver<ResolutionEvent>> {
```

Comment:
IMO `resolve` is generic, i.e. more than querying for addresses. Either we pass in an enum of different things to resolve, or define different methods for each kind of resolve. (Same point as I suggested to add a QueryType)